src/entities/DealRegistrationTokenSwap.ts

```diff
@@ -18,10 +18,18 @@ export enum Platforms {
 }
 
 export interface IToken {
-  name: string,
-  symbol: string,
-  balance: string,
   address: string,
+
+  name?: string,
+  symbol?: string,
+  decimals?: number,
+  logoURI?: string,
+
+  amount?: number;
+  instantTransfer: number
```

Comment:
amount, instantTransfer and vestedTransfer should be in wei, saved as a string (see, for example in launch, tokenDetails.maxSupply). Same for all eth/token amounts -- they must always be passed around in wei. numeric-input takes care of that, including converting to a string.

BTW I'd kinda prefer "instantTransferAmount" and "vestedTransferAmount"